### PR TITLE
Initial text block, with a mixin to generate quote image fields.

### PR DESCRIPTION
**Pull request by AnmAtAnm**

### CODE REVIEW

File: blocks/text.js [hunk 1]
```diff
@@ -776,3 +785,25 @@ Blockly.Blocks['text_reverse'] = {
     });
   }
 };
+
+/**
+ * Mixin to add a helper function that generates a FieldImage of an opening or
+ * closing double quote. The selected quote will be adapted for RTL blocks.
+ * @param {boolean} open If the image should be open (“ in LTR). Otherwise, a
+ *                       closing quote is used (” in LTR).
+ * @returns {!Blockly.FiledImage}
```
Comment (rachel-fenichel) [inline blocks/text.js]: type: FieldImage, not FiledImage

File: core/field_image.js [hunk 2]
```diff
@@ -43,6 +43,19 @@ goog.require('goog.userAgent');
  */
 Blockly.FieldImage = function(src, width, height, opt_alt) {
   this.sourceBlock_ = null;
+
+  // Replace any message references in the arguments.
+  src = Blockly.utils.replaceMessageReferences(src)
+  if (goog.isString(height)) {
+    height = Blockly.utils.replaceMessageReferences(height);
+  }
+  if (goog.isString(width)) {
```
Comment (rachel-fenichel) [inline core/field_image.js]: Why is the size a string?

Comment (AnmAtAnm) [inline core/field_image.js]: To allow references (acting as constants) in JSON, much like block color:
{
  "type": "field_image",
  "src": "%{BKY_OPENING_QUOTE_IMAGE}",
  "width": "%{BKY_QUOTE_IMAGE_SIZE}",
  "height": "%{BKY_QUOTE_IMAGE_SIZE}",
  "alt": "\u201C"
}

Alternative suggestions?
(corrected @param type)

File: blocks/text.js (hunk 1 above)
Comment (AnmAtAnm) [inline blocks/text.js]: Fixed.

**Comment by rachel-fenichel** [comment]
lgtm

**Comment by AnmAtAnm** [comment]
Major rewrite pre offline discussion. Text block now uses the extension "text_quotes", supported by Blockly.Constants.Text.QUOTE_IMAGE_MIXIN.quoteField_(fieldName), so that each platform can use the best platform appropriate image (size, density, etc.) for the quotes.

All other support for message references has been left in, though it is no longer used.

File: blocks/text.js [hunk 3]
```diff
@@ -776,3 +781,71 @@ Blockly.Blocks['text_reverse'] = {
     });
   }
 };
+
+/**
+ *
```
Comment (rachel-fenichel) [inline blocks/text.js]: Either an extra newline or missing doc here.

File: blocks/text.js [hunk 4]
```diff
@@ -776,3 +781,71 @@ Blockly.Blocks['text_reverse'] = {
...
+ * @mixin
+ * @package
+ * @readonly
+ */
+Blockly.Constants.Text.QUOTE_IMAGE_MIXIN = {
+  /**
+   * Image data URI of an LTR opening double quote (same as RTL closing couble quote).
```
Comment (rachel-fenichel) [inline blocks/text.js]: *double quote, not couble quote.  Here and below.

**Pull request merged**